how-to/customize-workspace/client/src/headless.ts

```diff
@@ -0,0 +1,45 @@
+import { createLogger } from "./logger-provider";
+import type { ModuleDefinition } from "./module-shapes";
+import type { HeadlessProviderOptions } from "./shapes";
+
+const logger = createLogger("HeadlessProvider");
+
+async function launchHeadlessEntry(headlessEntry: ModuleDefinition<OpenFin.WindowOptions>) {
+	try {
+		logger.info(`Launching headless window with id ${headlessEntry.id}`);
+		const windowOptions: OpenFin.WindowOptions & { backgroundThrottling?: boolean } = {
+			url: headlessEntry.url,
+			name: headlessEntry.id,
+			autoShow: false,
+			includeInSnapshots: false,
+			showTaskbarIcon: false,
+			backgroundThrottling: false,
+			...headlessEntry.data
+		};
+		return await fin.Window.create(windowOptions);
+	} catch (error) {
+		logger.error(`Error launching headless window with id ${headlessEntry.id}`, error);
+		return null;
+	}
+}
+export async function init(options?: HeadlessProviderOptions) {
+	if (options?.modules === undefined || options?.modules?.length === 0) {
+		logger.info("There are no settings for the headless provider. No headless logic required.");
+		return;
+	}
+
+	logger.info("Initializing Headless Provider", options);
+
+	const numberOfModules = options?.modules?.length ?? 0;
```

Comment:
Could this move to the top of the method instead of calculating it twice, or just use IsArray for the initial test and a `for...of` instead of a counter loop to improve readability

Reply:
updated